File: data/data.code.javascript.client/ReqTestJs.js
module.exports = ReqTestJs;

var Packet = require('../net/Packet');


function ReqTestJs() {
	this._u64 = 0;
	this._i64 = 0;


	this.Encode = function() {
		var packet = new Packet();
		packet.WriteUlong(this._u64);
		packet.WriteLong(this._i64);
		packet.Encode(40080);
		return packet;
	}
}

ReqTestJs.prototype = {
	set U64(val) {
		this._u64 = val;
	},
	get U64() {
		return this._u64;
	},

	set I64(val) {
		this._i64 = val;
	},
	get I64() {
		return this._i64;
	},
}

File: data/data.code.lua.client/ReqTestJs.lua
module("ReqTestJs", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqTestJs})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.u64 = 0
	self.i64 = 0
end


function encode(self)
	local pack = packet.new()
	pack:writeU64(self.u64)
	pack:writeI64(self.i64)
	return pack:encode(Msg.P_REQ_TEST_JS)
end

return ReqTestJs
